### PR TITLE
Adds Retweet Feature

### PR DESCRIPTION
**Pull request by kyljmn**
##Changes
-Adds retweet and unretweet functionality
-Displays retweets together with normal twits in Home page and profile page
-Indicates whether a tweet has already been retweeted by the current user by showing unretweet action
##Screenshots
tweeter_PR_feature_retweet_home
Home page displays retweets together with tweets

### CODE REVIEW

File: app/controllers/home_controller.rb [hunk 1]
```diff
@@ -3,9 +3,12 @@ def index
     if !user_signed_in?
       redirect_to new_user_session_path 
     else
+      @twit = Twit.new()
       @user = User.find(current_user.id)
       relevant_users = @user.following_ids + [@user.id]
-      @twits = Twit.where(user: relevant_users).reverse()
+      @twits = Twit.where(user: relevant_users).to_a
```
Comment (JTaylorGonzales) [inline app/controllers/home_controller.rb:9]: array na return nito a.

File: app/controllers/home_controller.rb [hunk 2]
```diff
@@ -3,9 +3,12 @@ def index
...
+      @retwits = Retwit.where(retwiter: relevant_users).to_a
+      @all = (@twits + @retwits).sort_by(&:created_at).reverse
```
Comment (JTaylorGonzales) [inline app/controllers/home_controller.rb:11]: wag nyo gawin tong mga ganto sa controllers. create a separate service for this. ex: `app/services/put_all_tweets` or something

File: app/views/twits/_table.html.erb [hunk 3]
```diff
@@ -6,6 +6,10 @@
     <th>Action</th>
   </tr>
   <% twits.map do |twit| %>
-    <%= render twit %>
+    <% if twit.class.name == "Twit" %>
+      <%= render twit %>
```
Comment (JTaylorGonzales) [inline app/views/twits/_table.html.erb:10]: i think pwede to i rewrite as `<%= render twit if twit.class.name == "Twit" %>` para mas maikli

File: app/views/twits/_twit.html.erb [hunk 4]
```diff
@@ -1,5 +1,16 @@
 <tr>
-  <td><%= twit.user.username %></td>  
+  <td><%= link_to twit.user.username, user_path(twit.user) %></td>  
   <td><%= twit.body %></td>
-  <td><%= link_to "x", user_twit_path(twit.user.id, twit), method: :delete %></td>
+  <td>
+    <% if twit.user.id == current_user.id %>
+      <%= link_to "x", user_twit_path(twit.user.id, twit), method: :delete %>
+    <% elsif !twit.retwits.exists?(retwiter: current_user) %>
```
Comment (JTaylorGonzales) [inline app/views/twits/_twit.html.erb:7]: wag gawin yung mga gantong logic sa template. lagay mo to sa view helper.

**Review by JTaylorGonzales — CHANGES_REQUESTED** [review]
left some comments:

1. avoid putting logic sa templates
2. same sa controller, put it in a service para ma reuse mo and to test it easily

File: app/controllers/home_controller.rb (hunk 1 above)
Comment (kyljmn) [reply app/controllers/home_controller.rb:9]: yup yup kasi kailangan ko ipagsama yung tweets at retweets sa isang array para masort na latest first

File: app/controllers/home_controller.rb (hunk 2 above)
Comment (kyljmn) [reply app/controllers/home_controller.rb:11]: ah okay okay

File: app/controllers/home_controller.rb (hunk 1 above)
Comment (JTaylorGonzales) [reply app/controllers/home_controller.rb:9]: no dont do that hahaha. pwede mo gawin sa SQL lang yan.

Comment (kyljmn) [reply app/controllers/home_controller.rb:9]: okie

File: app/controllers/home_controller.rb [hunk 5]
```diff
@@ -3,9 +3,10 @@ def index
     if !user_signed_in?
       redirect_to new_user_session_path 
     else
+      @twit = Twit.new()
       @user = User.find(current_user.id)
       relevant_users = @user.following_ids + [@user.id]
-      @twits = Twit.where(user: relevant_users).reverse()
+      @all = helpers.get_twits_retwits(relevant_users)
```
Comment (JTaylorGonzales) [inline app/controllers/home_controller.rb:9]: wag sa application helper i lagay. helpers is mostly for templates sya. yung sa controller gawa ka ng sarili mong service

**Pull request merged**